plugins/workspace-minimap/src/focus_mode.ts

```diff
@@ -0,0 +1,171 @@
+ */
+export class FocusMode {
+    private primaryWorkspace: Blockly.WorkspaceSvg;
+    private minimapWorkspace: Blockly.WorkspaceSvg;
+    private onChangeWrapper: (e: Blockly.Events.Abstract) => void;
+    private svgGroup: SVGElement;
+    private rect: SVGElement;
+    private background: SVGElement;
+    private id: string;
+
+
+    /**
+     * Constructor for focus mode.
+     * @param primary The primary workspaceSvg.
+     * @param minimap The minimap workspaceSvg.
+     */
+    constructor(primary: Blockly.WorkspaceSvg, minimap: Blockly.WorkspaceSvg) {
+      this.primaryWorkspace = primary;
+      this.minimapWorkspace = minimap;
+      this.id = this.minimapWorkspace.id;
+    }
+
+
+    /**
+     * Initializes focus mode.
+     */
+    init() {
+      // Make the svg group element.
+      this.svgGroup = Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.G, {'class': 'focusMode'}, null);
+
+      // Make the mask under the svg group.
+      const mask = Blockly.utils.dom.createSvgElement(
+          new Blockly.utils.Svg('mask'),
+          {'id': 'focusModeMask' + this.id},
+          this.svgGroup);
+
+      // Make the backround under the svg group.
+      this.background = Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'width': '100%',
+            'height': '100%',
+            'mask': 'url(#focusModeMask' + this.id + ')',
+          }, this.svgGroup);
+
+      // Make the white layer under the svg mask.
+      Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'width': '100%',
+            'height': '100%',
+            'fill': 'white',
+          }, mask);
+
+      // Make the black layer under the mask.
+      this.rect = Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'rx': Blockly.Bubble.BORDER_WIDTH,
+            'ry': Blockly.Bubble.BORDER_WIDTH,
+            'fill': 'black',
+          }, mask);
+
+      // Theme.
+      this.background.setAttribute('fill', '#e6e6e6');
+
+      // Add the svg group to the minimap.
+      const parentSvg = this.minimapWorkspace.getParentSvg();
+      if (parentSvg.firstChild) {
+        parentSvg.insertBefore(this.svgGroup, parentSvg.firstChild);
+      } else {
+        parentSvg.appendChild(this.svgGroup);
+      }
+
+      window.addEventListener('resize', () => void this.update());
+      this.onChangeWrapper = this.onChange.bind(this);
+      this.primaryWorkspace.addChangeListener(this.onChangeWrapper);
+
+      this.update();
+    }
+
+
+    /**
+     * Disposes of the focus mode.
+     * Unlinks from all DOM elements and remove all event listeners
+     * to prevent memory leaks.
+     */
+    dispose() {
+      if (this.svgGroup) {
+        Blockly.utils.dom.removeNode(this.svgGroup);
```

Comment:
You should also remove any references to the svg elements (`svgGroup`, `background`, and `rect`). If there are still references to them, they won't get garbage collected even though they've been disconnected from the dom.

You can remove references by setting the properties to null:
```
this.svgGroup = null;
```